core/src/main/java/org/pentaho/platform/scheduler2/action/SchedulerFilenameUtils.java

```diff
@@ -0,0 +1,99 @@
+ *
+ *
+ * Copyright (c) 2024 Hitachi Vantara. All rights reserved.
+ *
+ */
+
+package org.pentaho.platform.scheduler2.action;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * General filename and filepath manipulation utilities. Primarily to support Pentaho Repository and URI file paths.
+ * Similar logic to {@link org.pentaho.platform.repository.RepositoryFilenameUtils}
+ * and {@link org.apache.commons.io.FilenameUtils }
+ *
+ */
+public class SchedulerFilenameUtils {
+
+  /**
+   * Hiding default constructor this is a utility class.
+   */
+  private SchedulerFilenameUtils() {
+    // EMPTY ON PURPOSE
+  }
+
+  /**
+   * Separator for file paths. Assuming unix style separator due pentaho domain.
+   */
+  protected static final String PATH_SEPARATOR = "/";
+
+  /**
+   * Combine <code>basePath</code> and <code>pathToAdd</code> using path separator defined by {@link #PATH_SEPARATOR}.
+   * <p/>
+   *
+   * Examples:
+   * <p/>
+   * <pre>
+   * /foo + bar                  -->   /foo/bar
+   * /foo/ + bar                 -->   /foo/bar
+   * /foo + /bar                 -->   /foo/bar
+   * /foo/ + /bar                -->   /foo/bar
+   * /foo + bar.txt              -->   /foo/bar.txt
+   * /foo/ + bar.txt             -->   /foo/bar.txt
+   * /foo + /bar.txt             -->   /foo/bar.txt
+   * /foo/ + /bar.txt            -->   /foo/bar.txt
+   * scheme://foo + bar          -->   scheme://foo/bar
+   * scheme://foo/ + bar         -->   scheme://foo/bar
+   * scheme://foo + /bar         -->   scheme://foo/bar
+   * scheme://foo/ + /bar        -->   scheme://foo/bar
+   * scheme://foo + bar.txt      -->   scheme://foo/bar.txt
+   * scheme://foo/ + bar.txt     -->   scheme://foo/bar.txt
+   * scheme://foo + /bar.txt     -->   scheme://foo/bar.txt
+   * scheme://foo/ + /bar.txt    -->   scheme://foo/bar.txt
+   * </pre>
+   *
+   * @param basePath
+   * @param pathToAdd
+   * @return combined path consisting of: <code>basePath</code> + {@value #PATH_SEPARATOR} + <code>pathToAdd</code>
+   */
+  public static String concat( String basePath, String pathToAdd ) {
+
+    if ( basePath == null || pathToAdd == null ) {
+      return null;
+    }
+    /*
+     * NOTE don't alter schema syntax ie remove // in a URI.
+     * Don't call any function that eventually calls org.apache.commons.io.FilenameUtils#normalize
+     * Similar logic to org.pentaho.platform.web.http.api.resources.SchedulerOutputPathResolver
+     */
+    return getNoEndSeparator( basePath )  + PATH_SEPARATOR + FilenameUtils.getName( pathToAdd );
+  }
+
+  /**
+   * Return only the path without the ending separator defined by {@link #PATH_SEPARATOR}
+   * @param path
+   * @return <code>path</code> without the end separator
+   */
+  protected static String getNoEndSeparator( String path ) {
+    return  ( StringUtils.isNotBlank( path )
+        && path.length() - 1 == FilenameUtils.indexOfLastSeparator( path ) )
```

Comment:
How paranoid do we want to be about checking for a double separator?  Seems unlikely to be an issue to me.

Reply:
So we have two types of paths that we will support for output directories.
Easier to look at unit tests in https://github.com/pentaho/pentaho-scheduler-plugin-ee/pull/163/files
- testConcat_PVFS()
- testConcat_Repo()

Pentaho repository: concat("/home/admin", "someFile.*")
PVFS concat("pvfs://somBucket/someFolder/", "someFile.*")

I'm getting the first part or directory from UI, I'm assuming it is a valid Pentaho repository path with single slashes.